### ISSUE TITLE
2.3.0 no longer supports async functions for config

### ISSUE THREAD

**Issue by joeldenning** [rollup/rollup#3470]
- Rollup Version: 2.3.0
- Operating System (or Browser): OSX and Linux
- Node Version (if applicable):
- Link to reproduction (IMPORTANT, read below): See https://github.com/esm-bundle/react-dom/pull/44 and https://travis-ci.com/github/esm-bundle/react-dom/builds/156373881

### Expected Behavior

Async functions should be supported as rollup configs. This worked in 2.2.0, but appears not to work in 2.3.0.

```js
export default async () => ({
  input: './index.js',
  output: {...}
})
```

### Actual Behavior

```sh
yarn build
yarn run v1.22.0
$ rollup -c
[!] Error: Config file must export an options object, or an array of options objects
https://rollupjs.org/guide/en/#configuration-files
Error: Config file must export an options object, or an array of options objects
    at Object.error (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/shared/rollup.js:10120:30)
    at getConfigList (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/shared/loadConfigFile.js:543:26)
    at loadConfigFile (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/shared/loadConfigFile.js:488:12)
    at Object.loadAndParseConfigFile (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/shared/loadConfigFile.js:466:21)
    at getConfigs (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/bin/rollup:2032:39)
    at runRollup (/Users/joeldenning/code/esm-bundle-react-dom/node_modules/rollup/dist/bin/rollup:2013:43)
```

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by joeldenning** [comment]
Here's perhaps a clearer reproduction case - https://github.com/joeldenning/rollup-3470

**Comment by lukastaegert** [comment]
Sorry about that and thanks for spotting. Fix at #3472.

**Comment by joeldenning** [comment]
Thank you for the very quick fix!
